x-pack/plugin/ml/log-structure-finder/src/main/java/org/elasticsearch/xpack/ml/logstructurefinder/LogStructureUtils.java

```diff
@@ -140,19 +155,20 @@ private LogStructureUtils() {
 
         if (fieldValues.stream().anyMatch(value -> value instanceof Map)) {
             if (fieldValues.stream().allMatch(value -> value instanceof Map)) {
-                return Collections.singletonMap(MAPPING_TYPE_SETTING, "object");
+                return new Tuple<>(Collections.singletonMap(MAPPING_TYPE_SETTING, "object"), null);
             }
             throw new IllegalArgumentException("Field [" + fieldName +
                 "] has both object and non-object values - this is not supported by Elasticsearch");
         }
 
         if (fieldValues.stream().anyMatch(value -> value instanceof List || value instanceof Object[])) {
             // Elasticsearch fields can be either arrays or single values, but array values must all have the same type
-            return guessMapping(explanation, fieldName,
+            return guessMappingAndCalculateFieldStats(explanation, fieldName,
                 fieldValues.stream().flatMap(LogStructureUtils::flatten).collect(Collectors.toList()));
         }
 
-        return guessScalarMapping(explanation, fieldName, fieldValues.stream().map(Object::toString).collect(Collectors.toList()));
+        Collection<String> fieldValuesAsStrings = fieldValues.stream().map(Object::toString).collect(Collectors.toList());
+        return new Tuple<>(guessScalarMapping(explanation, fieldName, fieldValuesAsStrings), calculateFieldStats(fieldValuesAsStrings));
```

Comment:
I considered this but then thought it would be nice if the class could be used in other situations where we don't know the mapping in advance.  I guess we can revisit this based on future experiences.